samples/csharp/getting-started/Ranking_AutoML/Ranking/Ranking/Program.cs

```diff
@@ -0,0 +1,175 @@
+        const string ValidationDatasetUrl = "https://aka.ms/mlnet-resources/benchmarks/MSLRWeb10KValidate240kRows.tsv";
+        const string TestDatasetUrl = "https://aka.ms/mlnet-resources/benchmarks/MSLRWeb10KTest240kRows.tsv";
+
+        readonly static string InputPath = Path.Combine(AssetsPath, "Input");
+        readonly static string OutputPath = Path.Combine(AssetsPath, "Output");
+        readonly static string TrainDatasetPath = Path.Combine(InputPath, "MSLRWeb10KTrain720kRows.tsv");
+        readonly static string ValidationDatasetPath = Path.Combine(InputPath, "MSLRWeb10KValidate240kRows.tsv");
+        readonly static string TestDatasetPath = Path.Combine(InputPath, "MSLRWeb10KTest240kRows.tsv");
+        readonly static string ModelPath = Path.Combine(OutputPath, "RankingModel.zip");
+
+        private static uint ExperimentTime = 60;
+
+        private static RankingData sampleRankingData = null;
+
+        static void Main(string[] args)
+        {
+            var mlContext = new MLContext(seed: 0);
+
+            // Create, train, evaluate and save a model
+            BuildTrainEvaluateAndSaveModel(mlContext);
+
+            // Make a single test prediction loading the model from .ZIP file
+            TestSinglePrediction(mlContext);
+
+            Console.WriteLine("=============== End of process, hit any key to finish ===============");
+            Console.ReadKey();
+        }
+
+        private static ITransformer BuildTrainEvaluateAndSaveModel(MLContext mlContext)
+        {
+            // STEP 1: Download and load the data
+            GetData(InputPath, OutputPath, TrainDatasetPath, TrainDatasetUrl, TestDatasetUrl, TestDatasetPath,
+                ValidationDatasetUrl, ValidationDatasetPath);
+
+            IDataView trainDataView = mlContext.Data.LoadFromTextFile<RankingData>(TrainDatasetPath, hasHeader: true, separatorChar: '\t');
+            IDataView testDataView = mlContext.Data.LoadFromTextFile<RankingData>(TestDatasetPath, hasHeader: true, separatorChar: '\t');
+
+            // STEP 2: Display first few rows of training data
+            ConsoleHelper.ShowDataViewInConsole(mlContext, trainDataView);
+
+            // STEP 3: Initialize our user-defined progress handler that AutoML will 
+            // invoke after each model it produces and evaluates.
+            var progressHandler = new RankingExperimentProgressHandler();
+
+            // STEP 4: Run AutoML ranking experiment
+            ConsoleHelper.ConsoleWriteHeader("=============== Running AutoML experiment ===============");
+            Console.WriteLine($"Running AutoML ranking experiment for {ExperimentTime} seconds...");
+
+            ExperimentResult<RankingMetrics> experimentResult = mlContext.Auto()
+                .CreateRankingExperiment(ExperimentTime)
+                .Execute(trainDataView, progressHandler: progressHandler);
```

Comment:
Since it's available, I'd recommend handing it both the train and validation sets. 

When given a singular training set with out a validation set, it will split the training datasets (or use CV) to auto-create a validation set. But since we have one we should use it.